api/src/main/java/org/openmrs/Obs.java

```diff
@@ -1072,7 +1072,11 @@ public String getValueAsString(Locale locale) {
 	public void setValueAsString(String s) throws ParseException {
 		log.debug("getConcept() == {}", getConcept());
 		
-		if (getConcept() != null && !StringUtils.isBlank(s)) {
+		if (StringUtils.isEmpty(s)) {
+			throw new RuntimeException("Value cannot be null or empty");
+		}
```

Comment:
Changed this from isBlank to isEmpty and moved into a separate block (since previously the exception message was incorrect)